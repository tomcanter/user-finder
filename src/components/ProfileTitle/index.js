// @flow

import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';
import 'suitcss-utils-flex/lib/flex.css';

type Props = {
  userLink: string,
  username: string,
  name: string,
};

export default function ProfileTitle(props: Props) {
  const {
    name,
    username,
    userLink,
  } = props;

  return (
    <div className="u-flex u-flexWrap u-flexAlignItemsBaseline">
      <h1 className={css(styles.ProfileTitle_name)}>{name}</h1>
      <p className={css(styles.ProfileTitle_username)}>
        <a href={userLink}>@{username}</a>
      </p>
    </div>
  );
}

const styles = StyleSheet.create({
  ProfileTitle_username: {
    color: '#555',
  },

  ProfileTitle_name: {
    marginRight: 5,
    fontSize: 28,
  },
});

