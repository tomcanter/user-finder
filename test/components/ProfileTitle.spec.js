import React from 'react';
import renderer from 'react-test-renderer';
import {StyleSheetTestUtils} from 'aphrodite';
import ProfileTitle from 'components/ProfileTitle';

describe('Component: ProfileTitle', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(<ProfileTitle
      name="Tom"
      username="tomcanter"
      userLink="http://tomcanter.io"
    />);
    expect(component).toMatchSnapshot();
  });

});
