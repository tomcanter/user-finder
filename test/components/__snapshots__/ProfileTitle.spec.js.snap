// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProfileTitle should render 1`] = `
<div
  className="u-flex u-flexWrap u-flexAlignItemsBaseline"
>
  <h1
    className="ProfileTitle_name_1oubu48"
  >
    Tom
  </h1>
  <p
    className="ProfileTitle_username_irnxte"
  >
    <a
      href="http://tomcanter.io"
    >
      @
      tomcanter
    </a>
  </p>
</div>
`;
